### PR TITLE
Make getIdeaInstallDir more generic to support other platform types

### PR DESCRIPTION
**Pull request by mkondratek**
## Test plan
1. runIde with pycharm and with idea

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -409,12 +409,23 @@ tasks {
           "cody.autocomplete.enableFormatting" to
               (project.property("cody.autocomplete.enableFormatting") ?: "true"))
 
-  fun getIdeaInstallDir(ideaVersion: String, ideaType: String): File? {
+  fun getPlatformName(ideaType: String) =
```
Comment (pkukielka) [inline build.gradle.kts:412]: https://github.com/JetBrains/intellij-platform-gradle-plugin/blob/main/src/main/kotlin/org/jetbrains/intellij/platform/gradle/IntelliJPlatformType.kt

Comment (mkondratek) [reply build.gradle.kts:412]: Not yet available with 1.17.4, we must move to the new gradle plugin (v2).

Comment (pkukielka) [reply build.gradle.kts:412]: Maybe just after launch it will be good time to do that anyway?

**Review by dominiccooney — APPROVED** [review]
Being able to test more IDEs sounds great.

- Please update the CONTRIBUTING.md documenting how to use this.
- Why did you stop at just pycharm? Could you add Goland, we seem to get a bunch of issues from Goland users. Ideally a bunch of 'em.
- Upgrading the build infra sounds good.

**Comment by mkondratek** [comment]
It looks like we do not need these changes with the new intellij gradle platform plugin 2. closing!

**Pull request closed without merging**